src/Microsoft.AspNetCore.Razor.Language/LargeTextSourceDocument.cs

```diff
@@ -17,8 +18,9 @@ internal class LargeTextRazorSourceDocument : RazorSourceDocument
         private readonly RazorSourceLineCollection _lines;
 
         private readonly int _length;
+        private byte[] _checksum;
 
-        public LargeTextRazorSourceDocument(StreamReader reader, int chunkMaxLength, Encoding encoding, string fileName)
+        public LargeTextSourceDocument(StreamReader reader, int chunkMaxLength, Encoding encoding, string fileName)
```

Comment:
Does this actually work if we have a streamreader? Does the streamreader normalize the BOM?